lib/membrane/rtp/ssrc_router.ex

```diff
@@ -286,13 +305,53 @@ defmodule Membrane.RTP.SSRCRouter do
     )
   end
 
-  defp emit_packet_arrival_event(payload, pad, ctx) do
+  defp register_inbound_frame_event(pad, ctx) do
+    Membrane.TelemetryMetrics.register(
+      @marker_received_telemetry_event,
+      ctx.pads[pad].options.telemetry_label
+    )
+  end
+
+  defp emit_rtcp_arrival_event(destination, ctx) do
+    Membrane.TelemetryMetrics.execute(
+      @rtcp_arrival_event,
+      %{},
+      %{},
+      ctx.pads[destination].options.telemetry_label
+    )
+  end
+
+  defp emit_rtcp_sent_event(destination, ctx) do
+    Membrane.TelemetryMetrics.execute(
+      @rtcp_sent_event,
+      %{},
+      %{},
+      ctx.pads[destination].options.telemetry_label
+    )
+  end
+
+  defp emit_packet_arrival_events(actions, ctx) do
+    for action <- actions do
+      with {:buffer, {pad, buffer}} <- action do
+        emit_packet_arrival_event(buffer, pad, ctx)
+      end
+    end
+  end
+
+  defp emit_packet_arrival_event(%Buffer{} = buffer, pad, ctx) do
+    packet_size = byte_size(buffer.payload)
+
+    label = ctx.pads[pad].options.telemetry_label
+
     Membrane.TelemetryMetrics.execute(
       @packet_arrival_event,
-      %{bytes: byte_size(payload)},
+      %{bytes: packet_size},
       %{},
-      ctx.pads[pad].options.telemetry_label
+      label
     )
+
+    if packet_size == 254,
+      do: Membrane.TelemetryMetrics.execute(@padding_packet_arrival_event, %{}, %{}, label)
```

Comment:
IIRC this is not the correct way to detect padding packets, it will only work for encrypted packets, regular RTP will have the padding stripped

Reply:
Actually, it will work a little too well :facepalm: It will also count non-padding packets that happen to have that exact size

Reply:
I think we should drop this metric for now and get this Pull Request through without it.

Given that the size is not enough to determine if the packet is a padding packet **AND** the padding also seems to be encrypted, we need something that lives after the decryptor to detect padding packets.
I'm also not convinced it would be a very useful metric on its own in the RTP plugin outside of WebRTC, so I'm considering implementing it in the RTC Engine. Buffers that pass through RTC Engine are decrypted and they have their headers removed, so it should be a simple matter of `byte_size(packet) == 0 || packet == <<0::binary-size(254), 255::8>>`

Reply:
>I'm also not convinced it would be a very useful metric on its own in the RTP plugin outside of WebRTC, so I'm considering implementing it in the RTC Engine.

This sounds like a good idea, padding packets seem specific to WebRTC, but probably not possible without refactoring - after inbound RTX has been added, the padding packets are sent with retransmission SSRC and RTXParser drops them (they cannot be parsed as they do not contain a payload with the original sequence number)